frame/treasury/src/lib.rs

```diff
@@ -124,10 +295,183 @@ decl_module! {
 		/// Percentage of spare funds (if any) that are burnt per spend period.
 		const Burn: Permill = T::Burn::get();
 
+		/// The period for which a tip remains open after is has achieved threshold tippers.
+		const TipCountdown: T::BlockNumber = T::TipCountdown::get();
+
+		/// The amount of the final tip which goes to the original reporter of the tip.
+		const TipFindersFee: Percent = T::TipFindersFee::get();
+
+		/// The amount held on deposit for placing a tip report.
+		const TipReportDepositBase: BalanceOf<T> = T::TipReportDepositBase::get();
+
+		/// The amount held on deposit per byte within the tip report reason.
+		const TipReportDepositPerByte: BalanceOf<T> = T::TipReportDepositPerByte::get();
+
 		type Error = Error<T>;
 
 		fn deposit_event() = default;
 
+		/// Report something `reason` that deserves a tip and claim any eventual the finder's fee.
+		///
+		/// The dispatch origin for this call must be _Signed_.
+		///
+		/// Payment: `TipReportDepositBase` will be reserved from the origin account, as well as
+		/// `TipReportDepositPerByte` for each byte in `reason`.
+		///
+		/// - `reason`: The reason for, or the thing that deserves, the tip; generally this will be
+		///   a UTF-8-encoded URL.
+		/// - `who`: The account which should be credited for the tip.
+		///
+		/// Emits `NewTip` if successful.
+		///
+		/// # <weight>
+		/// - `O(R)` where `R` length of `reason`.
+		/// - One balance operation.
+		/// - One storage mutation (codec `O(R)`).
```

Comment:
When you say `codec `O(R)` you are describing that interacting with this storage item will require us to parity-scale-codec encode/decode R items?

Reply:
correct.